File: backend/src/modules/siptrace/siptrace.controller.ts
import { Controller, Get, Post, Body, Query } from '@nestjs/common';
import { ApiTags, ApiOperation } from '@nestjs/swagger';
import { SiptraceService, SipTraceQueryDto } from './siptrace.service';
import { ApiResponseDto } from '../../common/dto/api-response.dto';

@ApiTags('SIP 跟踪')
@Controller('api/siptrace')
export class SiptraceController {
  constructor(private readonly siptraceService: SiptraceService) {}

  @Get()
  @ApiOperation({ summary: '获取 SIP 跟踪记录' })
  async findAll(@Query() dto: SipTraceQueryDto) {
    return ApiResponseDto.success(await this.siptraceService.findAll(dto));
  }

  @Get('call')
  @ApiOperation({ summary: '获取呼叫的完整跟踪' })
  async getByCallId(@Query('callid') callid: string) {
    return ApiResponseDto.success(await this.siptraceService.getByCallId(callid));
  }

  @Get('stats')
  @ApiOperation({ summary: '获取统计' })
  async getStats() {
    return ApiResponseDto.success(await this.siptraceService.getStats());
  }

  @Post('cleanup')
  @ApiOperation({ summary: '清理旧记录' })
  async cleanup(@Body() body: { days: number }) {
    const beforeDate = new Date();
    beforeDate.setDate(beforeDate.getDate() - (body.days || 30));
    const count = await this.siptraceService.cleanup(beforeDate);
    return ApiResponseDto.success({ deleted: count }, `清理了 ${count} 条记录`);
  }
}

File: kam-admin-module-trace/src/main/java/io/kamailio/admin/modules/siptrace/controller/SiptraceController.java
package io.kamailio.admin.modules.siptrace.controller;

import io.kamailio.admin.common.dto.ApiResponse;
import io.kamailio.admin.modules.siptrace.entity.SipTrace;
import io.kamailio.admin.modules.siptrace.service.SiptraceService;
import io.kamailio.admin.common.dto.PaginatedResult;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.time.Instant;
import java.time.temporal.ChronoUnit;
import java.util.List;
import java.util.Map;

@Tag(name = "SIP 跟踪")
@RestController
@RequestMapping("/api/siptrace")
public class SiptraceController {
    private final SiptraceService service;

    public SiptraceController(SiptraceService service) { this.service = service; }

    @GetMapping
    @Operation(summary = "获取 SIP 跟踪记录")
    public ResponseEntity<ApiResponse<PaginatedResult<SipTrace>>> findAll(
            @RequestParam(required = false) Integer page,
            @RequestParam(required = false) Integer limit,
            @RequestParam(required = false) String callid,
            @RequestParam(required = false) String tracedUser,
            @RequestParam(required = false) String method,
            @RequestParam(required = false) String fromIp,
            @RequestParam(required = false) String toIp,
            @RequestParam(required = false) String startTime,
            @RequestParam(required = false) String endTime) {
        return ResponseEntity.ok(ApiResponse.success(service.findAll(page, limit, callid, tracedUser, method, fromIp, toIp, startTime, endTime)));
    }

    @GetMapping("/call")
    @Operation(summary = "获取呼叫的完整跟踪")
    public ResponseEntity<ApiResponse<List<SipTrace>>> getByCallId(@RequestParam String callid) {
        return ResponseEntity.ok(ApiResponse.success(service.getByCallId(callid)));
    }

    @GetMapping("/stats")
    @Operation(summary = "获取统计")
    public ResponseEntity<ApiResponse<Map<String, Object>>> getStats() {
        return ResponseEntity.ok(ApiResponse.success(service.getStats()));
    }

    @PostMapping("/cleanup")
    @Operation(summary = "清理旧记录")
    public ResponseEntity<ApiResponse<Map<String, Object>>> cleanup(@RequestBody Map<String, Integer> body) {
        int days = body != null && body.containsKey("days") ? body.get("days") : 30;
        Instant beforeDate = Instant.now().minus(days, ChronoUnit.DAYS);
        int count = service.cleanup(beforeDate);
        return ResponseEntity.ok(ApiResponse.success(Map.of("deleted", count), "清理了 " + count + " 条记录"));
    }
}
